spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java

```diff
@@ -209,15 +210,19 @@ private int batchSize(boolean isParquetOnly, boolean isOrcOnly) {
 
   @Override
   public Statistics estimateStatistics() {
+    return estimateStatistics(table.currentSnapshot());
+  }
+
+  protected Statistics estimateStatistics(Snapshot snapshot) {
     // its a fresh table, no data
-    if (table.currentSnapshot() == null) {
+    if (snapshot == null) {
       return new Stats(0L, 0L);
     }
 
     // estimate stats using snapshot summary only for partitioned tables (metadata tables are unpartitioned)
     if (!table.spec().isUnpartitioned() && filterExpressions.isEmpty()) {
       LOG.debug("using table metadata to estimate table statistics");
-      long totalRecords = PropertyUtil.propertyAsLong(table.currentSnapshot().summary(),
+      long totalRecords = PropertyUtil.propertyAsLong(snapshot.summary(),
```

Comment:
Is this an unrelated fix?

Reply:
It is related. I just forgot to invoke it from `SparkCopyOnWriteScan`.

Reply:
Should have the missing part now.